server/controllers/registration.controller.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const responseUtil = require('../utils/responseUtils');
+const MessageUtil = require('../utils/messageUtil');
+const userRegistrationService = require('../services/registration.service');
+const userServices = require('../services/user.service')
+const { sendCourseRegistrationMail } = require('../utils/emailUtils');
+const { v4: uuidv4 } = require('uuid');
+
+exports.registerUserToCourse = async (req, res) => {
+    const {
+        candidateName,
+        courseCode,
+        userId,
+    } = req.body;
+    const id = uuidv4();
+    try {
+        const courseData = {
+            id,
+            candidateName,
+            courseCode,
+            userId,
+            registrationStatus: "PENDING"
+        };
```

Comment:
for registering courses, I think we need `courseId`  instead of `courseCode` as admin has the provision to update `courseCode` anytime, so it cannot be considered as key.

Reply:
Updated to courseId for University and Course Registration Controllers